File: Backend/src/modules/tooltip/material/character_tooltip.rs
use crate::modules::tooltip::domain_value::{CharacterGuild, CharacterTooltipItem};

#[derive(Debug, Clone, Serialize, JsonSchema)]
pub struct CharacterTooltip {
    pub name: String,
    pub server: String,
    pub guild: Option<CharacterGuild>,
    pub faction: bool,
    pub hero_class_id: u8,
    pub expansion_id: u8,
    pub race_id: u8,
    pub gender: bool,
    pub level: u8,
    pub items: Vec<CharacterTooltipItem>,
}

File: Webclient/src/app/module/tooltip/module/character_tooltip/domain_value/character_tooltip.ts
export interface CharacterTooltip {
    guild: { name: string; rank: string; };
    name: string;
    server: string;
    faction: boolean;
    hero_class_id: number;
    expansion_id: number;
    race_id: number;
    gender: boolean;
    level: number;
    items: Array<{ name: string; quality: number; }>;
}
